### PR TITLE
Watch

### PR DESCRIPTION
**Pull request by tehpsalmist**
This may be somewhat of a first iteration, but I believe this addresses the concerns regarding not pounding the API and also giving the user a good, quick experience of deployment. Issue #6 will also enhance the user experience by speeding up the deploy time.

The current handling of multiple concurrent-ish save events is like this: the deploy command begins running on the first event (quick response--I think this is important), and any subsequent save events while the deploy is in progress are deflected, but another flag is set telling the app to run the deploy command again when it is finished to pick up any of the additional changes. This is a double deploy step, but it is done sequentially (not concurrently), and _only_ twice, regardless of how many save events happen during the deploy process. I hope that explanation is helpful.

### CODE REVIEW

**Comment by alexweber** [comment]
@tehpsalmist looks good for a first pass man!  i'd like to see a more robust debouncing implementation for the final thing....  maybe this would be a good candidate for a `@beta` release... or maybe not since its a new command and won't break existing stuff so whatever :)  Let's get @AParks to sign off then anyone can merge

**Comment by tehpsalmist** [comment]
@alexweber can you expound on what might qualify as "more robust"? Is it the method of implementation (code style, basically), or the actual timing that you think could be tweaked?

**Comment by alexweber** [comment]
@tehpsalmist hey man by more robust i just mean maybe consider a "proper" debounce implementation like one of the below for example:
- https://davidwalsh.name/function-debounce
- https://www.npmjs.com/package/debounce
- https://github.com/rubenv/angular-debounce

Your solution is a great first pass and works fine, maybe it will hold up forever, who knows, but it's a tad simplistic/naive compared to some of those existing implementations that what I meant.... :)

**Comment by tehpsalmist** [comment]
Hey, I'm in the middle of putting in a 50ms debounce, so hang on to this for a minute.

**Comment by tehpsalmist** [comment]
Ok, she's good to go from what I can tell. Maybe someone disagrees with my decision to go with 50ms, but I personally appreciate the speed. Really the debounce is there to protect against instances where the machine itself (IDE, file system, whatever) initiates multiple change events. Most of us aren't manually hitting the save command twice in under 50ms without concentration and effort. The initial mechanism I built in should suffice for that case, which simply captures any concurrent save events during deployment and redeploys the plugin one more time after the first deploy is complete (thus catching any changes and deploying the latest edition of your code.

Bottom line: no matter how hard you and your IDE try to abuse the save/update/delete/move file commands, the Zengine API will not suffer for it, and when the dust clears, the code on your machine will match the code in the cloud. Win win.

**Comment by evert0n** [comment]
Just a suggestion I see the watch command is only deploying would be cool at some point if it was a param that can be chosen what it does which could be a deploy or a publish

**Comment by evert0n** [comment]
Regarding your note about the 50ms it also could be a param/argument and 50ms the default

**Comment by tehpsalmist** [comment]
Yes, I agree about that Everton, although I’m not sure how many people would be trying to have their plugin auto-published on every save. Still, that could be done later. Perhaps a —backend support is the next feature, or the denounce time param.

**Comment by tehpsalmist** [comment]
@alexweber @AParks @evert0n You folks down for merging? I'm loving it over here. Can't wait to share the fun with you all.

**Comment by evert0n** [comment]
LGTM 👍
